lib/trento/activity_logging/activity_catalog.ex

```diff
@@ -48,35 +33,222 @@ defmodule Trento.ActivityLog.ActivityCatalog do
     @cluster_checks_execution_request => {:cluster_checks_execution_request, 202}
   }
 
+  # Make sure trento application is loaded so that we can safely access its modules
+  Application.ensure_loaded(:trento)
+
+  # The following code extracts the domain event activities from the event modules found in trento application.
+  # Event modules are expected to be in the form of `Trento.{ResourceType}.Events.{EventName}`, for instance:
+  #   - Trento.Hosts.Events.HostRegistered
+  #   - Trento.Clusters.Events.ChecksSelected
+  #   - Trento.SapSystems.Events.SapSystemDeregistered
+  #   - Trento.Databases.Events.DatabaseHealthChanged
+  #
+  # For an event to be taken into account, the <ResourceType> in the second position of the module name
+  # must be one of the following:
+  #   - Hosts
+  #   - Clusters
+  #   - SapSystems
+  #   - Databases
+  #
+  # Any event module that lives in a directory containing the word `legacy` is ignored.
+  # This check is needed to avoid naming clashes with the valid event modules.
+  #
+  # Once the list of the relevant event modules is extracted, it gets transformed to a map with the following shape:
+  # %{
+  #   <EventModule>: {<EventActivityType>, <WhenOccurrenceIsInteresting>}
+  # }
+  #
+  # <EventActivityType> is the lowercase and underscored version of the last part of the event module name.
+  # <WhenOccurrenceIsInteresting> is anything useful to mark as interesting an activity occurrence.
+  # For connection related activities it is the status code that makes the occurrence interesting for logging.
+  #
+  # Marking <WhenOccurrenceIsInteresting> as `:always` means the occurrence is always interesting for logging.
+  # Currently, all events are marked as `:always` interesting for logging.
+  #
+  # example:
+  # %{
+  #   Trento.Hosts.Events.HostRegistered: {:host_registered, :always},
+  #   Trento.Clusters.Events.ChecksSelected: {:checks_selected, :always},
+  #   Trento.SapSystems.Events.SapSystemDeregistered: {:sap_system_deregistered, :always},
+  #   Trento.Databases.Events.DatabaseHealthChanged: {:database_health_changed, :always},
+  #   ...
+  # }
+  #
+  # Note that this is the same shape as the @connection_activities map.
+
+  quote do
+    unquote(
+      @domain_event_activities (case :application.get_key(:trento, :modules) do
```

Comment:
In total honesty, I am not completely happy with this fancy magic here.
Yes, it does what it's meant to do, however a bit meh maybe.

The alternatives would be getting rid of some quote/unquote and rely on functions (macros at most) or rely on a more static reference list of domain events by having either a list like the following
```elixir
@domain_events [
  :login_attempt,
  :resource_tagging,
  :resource_untagging,
  :api_key_generation,
  :saving_suma_settings,
  :changing_suma_settings,
  :clearing_suma_settings,
  :user_creation,
  :user_modification,
  :user_deletion,
  :profile_update,
  :cluster_checks_execution_request,
  :heartbeat_succeeded,
  :heartbeat_failed,
  :host_checks_health_changed,
  :host_checks_selected,
  :host_deregistered,
  :host_deregistration_requested,
  :host_details_updated,
  :host_health_changed,
  :host_registered,
  :host_restored,
  :host_rolled_up,
  :host_rollup_requested,
  :host_saptune_health_changed,
  :host_tombstoned,
  :provider_updated,
  :saptune_status_updated,
  :sles_subscriptions_updated,
  :software_updates_discovery_cleared,
  :software_updates_discovery_requested,
  :software_updates_health_changed,
  :checks_selected,
  :cluster_checks_health_changed,
  :cluster_deregistered,
  :cluster_details_updated,
  :cluster_discovered_health_changed,
  :cluster_health_changed,
  :cluster_registered,
  :cluster_restored,
  :cluster_rolled_up,
  :cluster_rollup_requested,
  :cluster_tombstoned,
  :host_added_to_cluster,
  :host_removed_from_cluster,
  :application_instance_deregistered,
  :application_instance_health_changed,
  :application_instance_marked_absent,
  :application_instance_marked_present,
  :application_instance_moved,
  :application_instance_registered,
  :sap_system_database_health_changed,
  :sap_system_deregistered,
  :sap_system_restored,
  :sap_system_rolled_up,
  :sap_system_rollup_requested,
  :sap_system_tombstoned,
  :sap_system_updated,
  :database_deregistered,
  :database_health_changed,
  :database_instance_deregistered,
  :database_instance_health_changed,
  :database_instance_marked_absent,
  :database_instance_marked_present,
  :database_instance_registered,
  :database_instance_system_replication_changed,
  :database_registered,
  :database_restored,
  :database_rolled_up,
  :database_rollup_requested,
  :database_tenants_updated,
  :database_tombstoned
]
```
or a bunch of references to the event modules
```elixir
Trento.Hosts.Events.ProviderUpdated
Trento.Hosts.Events.HostDeregistered
Trento.Databases.Events.DatabaseHealthChanged
Trento.Clusters.Events.ClusterTombstoned
Trento.Clusters.Events.ClusterDeregistered
Trento.Hosts.Events.HostSaptuneHealthChanged
Trento.Hosts.Events.HostChecksSelected
Trento.Clusters.Events.ClusterHealthChanged
Trento.Clusters.Events.ClusterChecksHealthChanged
Trento.Hosts.Events.HostRegistered
Trento.Hosts.Events.HostDetailsUpdated
Trento.Clusters.Events.ChecksSelected
Trento.Databases.Events.DatabaseTenantsUpdated
Trento.Clusters.Events.HostRemovedFromCluster
Trento.Hosts.Events.SlesSubscriptionsUpdated
Trento.SapSystems.Events.ApplicationInstanceMarkedPresent
Trento.SapSystems.Events.SapSystemRestored
Trento.Databases.Events.DatabaseInstanceSystemReplicationChanged
Trento.SapSystems.Events.SapSystemDatabaseHealthChanged
Trento.SapSystems.Events.ApplicationInstanceHealthChanged
Trento.Hosts.Events.HostHealthChanged
Trento.SapSystems.Events.SapSystemHealthChanged
Trento.Hosts.Events.SoftwareUpdatesHealthChanged
Trento.Clusters.Events.ClusterDiscoveredHealthChanged
Trento.SapSystems.Events.SapSystemUpdated
Trento.SapSystems.Events.SapSystemRolledUp
Trento.Hosts.Events.HeartbeatFailed
Trento.Hosts.Events.HostRollUpRequested
Trento.Clusters.Events.ClusterDetailsUpdated
Trento.Clusters.Events.ClusterRollUpRequested
Trento.SapSystems.Events.ApplicationInstanceDeregistered
Trento.Clusters.Events.HostAddedToCluster
# and more
```
open to suggestions :smile: